src/js/components/pages/scene-editor-gui.jsx

```diff
@@ -0,0 +1,793 @@
+var React = require('react');
+var _ = require('lodash');
+var SceneStore = require('../../stores/scene-store');
+var GridStore = require("../../stores/grid-store");
+var HubSendActions = require('../../actions/hub-send-actions');
+var SceneActions = require('../../actions/scene-actions');
+var Rnd = require('react-rnd');
+var FontAwesome = require('react-fontawesome');
+var Rectangle = require('react-rectangle');
+var ImageLoader = require('react-imageloader');
+
+var saveTimeout = null;
+
+var SceneEditorGUI = React.createClass({
+
+    getInitialState: function () {
+        return _.extend(this.getStateFromStores(), {
+            focusedMediaObject: null,
+            scene: null,
+            placement: null,
+            mediaObject: null,
+            mediaTypeSupported: null,
+            shouldSave: false,
+            height: 0,
+            width: 0,
+        });
+    },
+
+    getStateFromStores: function () {
+        return {
+            scene: SceneStore.getScene(this.props._id),
+        };
+    },
+
+    componentDidMount: function () {
+        console.log("SceneEditorGUI: Mounted")
+
+        //subscribe to changes
+        SceneStore.addChangeListener(this._onChange);
+        GridStore.addChangeListener(this._onFocusChange)
+
+        //this is the first point where rendered size of area can be captured
+        this.setState({width: this.SceneLayoutArea.offsetWidth, height: this.SceneLayoutArea.offsetHeight})
+    },
+
+    componentWillReceiveProps: function (nextProps) {
+        //same as focus change, same handling procedure, required because selected scene change dosn't appear to reliably fire change event?? 
+        //this._onFocusChange();
+    },
+
+    componentWillUnmount: function () {
+        if (saveTimeout) {
+            //this.saveToScene()
+            clearTimeout(saveTimeout);
+        }    
+        SceneStore.removeChangeListener(this._onChange);
+        GridStore.removeChangeListener(this._onFocusChange)
+    },
+
+    _onChange: function () {
+        //scene changed, clear all state
+        log("SceneEditorGUI: Scene change", this.getStateFromStores())
+        this.setState(this.getStateFromStores());
+    },
+
+    _onFocusChange: function () {
+
+        focusedMediaObject = GridStore.getFocusedMediaObject();
```

Comment:
Undeclared variables in a few places, they need to be declared so that they do not bubble up outside of the function, if this is not done carefully it can cause variable overrides and unexpected bugs.

Reply:
Will fix this